src/Controller/OrderController.php

```diff
@@ -576,4 +577,34 @@ public function continueAction(Request $request,
 
         return $this->redirectToRoute('restaurant', ['id' => $restaurants->first()->getId()]);
     }
+
+    /**
+     * @Route("/order/{hashid}/preview", name="order_preview")
+     */
+    public function dataPreviewAction($hashid, OrderRepository $orderRepository)
```

Comment:
Maybe this could be moved to being managed by API Platform? 
This would need a [Custom Item Data Provider](https://api-platform.com/docs/core/data-providers/#custom-item-data-provider) to load an order via its hashid.

Reply:
Thanks for suggestion. Yes I agree about to move this and be managed by API Platform.
But I don't fully understand how Customer Item Data Provider works.
I've created a data provider but it tries to get an order item  if a do a GET to `api/orders/LnOpkxEN6xE5M7XR` or `api/orders/51` (and I guess we just want the data provider to manage the hash id case, right?).
BTW, when I send a hash id like here  `api/orders/LnOpkxEN6xE5M7XR` in the data provider the param value is 0 instead of the hash. But if I send an ID like '51' the param is 51.

This is what I have:
```
   final class OrderDataProvider implements ItemDataProviderInterface, RestrictedDataProviderInterface
   {
      public function __construct(string $secret)
      {
          $this->hashids = new Hashids($secret, 16);
      }
  
      public function supports(string $resourceClass, string $operationName = null, array $context = []): bool
      {
          return Order::class === $resourceClass;
      }
  
      public function getItem(string $resourceClass, $hashid, string $operationName = null, array $context = []): ?Order
      {
          $decoded = $this->hashids->decode($hashid);
  
          if (count($decoded) !== 1) {
              throw new BadRequestHttpException(sprintf('Hashid "%s" could not be decoded', $hashid));
          }
  
          $id = current($decoded);
          return new Order($id);
      }
  }
```

At `services.yml`
```
AppBundle\Api\DataProvider\OrderDataProvider:
    arguments:
      $secret: '%secret%'
    tags:
      - { name: 'api_platform.item_data_provider', priority: 2 }
    autoconfigure: false
```

Reply:
No problem, I never implemented a Custom Item Data Provider in this use case.